web3/middleware/cache.py

```diff
@@ -60,20 +60,21 @@
 }
 
 
-def should_cache(method, params, response):
-    if 'error' in response:
-        return False
-    elif 'result' not in response:
-        return False
-
-    if response['result'] is None:
-        return False
+def _should_cache(method, params, response=None):
+    if response is not None:
```

Comment:
If the whole response is None, we should cache it? Not sure how that would happen, but it doesn't feel right.

Reply:
Ah, this is the pre request/post-request should I cache this check to prevent recursion.

- `response` is `None` when it's being called prior to making the request
- `response` is not `None` when checking if the response should be cached.

Having written this out, I'm now annoyed with the *messyness* of this API and would be happy for suggestions on cleaning it up.  Basically, there needs to be a way to say, don't even generate a response for requests which match this pattern.

Reply:
Alright, I actually chose to remove this and special case the check for whether the request is for the 'latest' block since that request should **never** be cached by this middleware.